src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -208,6 +292,82 @@ public static bool TryParseInt64(string value, out long result)
             return long.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
         }
 
+        internal static unsafe bool TryParseInt32(StringSegment value, out int result)
```

Comment:
Yea not sure why the old `ParseInt32` is internal but I kept the same encapsulation. Should we make that public too?